src/intelligence_layer/core/model.py

```diff
@@ -406,3 +408,135 @@ def to_instruct_prompt(
+    content: str
+
+
+class ChatModel(ABC, AlephAlphaModel):
+    RECOMMENDED_MODELS: ClassVar[list[str]] = []
+
+    def __init__(
+        self, name: str, client: AlephAlphaClientProtocol | None = None
+    ) -> None:
+        if name not in self.RECOMMENDED_MODELS or name == "":
+            warnings.warn(
+                "The provided model is not a recommended model for this model class."
+                "Make sure that the model you have selected is suited to be use for the prompt template used in this model class."
+            )
+        super().__init__(name, client)
+
+    @property
+    @abstractmethod
+    def eot_token(self) -> str:
+        pass
+
+    @abstractmethod
+    def to_chat_prompt(
+        self,
+        messages: list[Message],
+        response_prefix: Optional[str] = None,
+    ) -> RichPrompt:
+        pass
+
+
+class Pharia1ChatModel(ChatModel):
+    """Instruction model to be used for all `Pharia-1-LLM-*` models.
+
+    Args:
+        name: The name of a valid llama-3 model.
+            Defaults to `llama-3-8b-instruct`
+        client: Aleph Alpha client instance for running model related API calls.
+            Defaults to :class:`LimitedConcurrencyClient`
+    """
+
+    CHAT_PROMPT_TEMPLATE = PromptTemplate(
+        """<|begin_of_text|>{% for message in messages %}<|start_header_id|>{{message.role}}<|end_header_id|>
+
+{% promptrange instruction %}{{message.content}}{% endpromptrange %}<|eot_id|>{% endfor %}<|start_header_id|>assistant<|end_header_id|>
+
+{% if response_prefix %}{{response_prefix}}{% endif %}"""
+    )
+
+    RECOMMENDED_MODELS: ClassVar[list[str]] = [
+        "Pharia-1-LLM-7b-control",
+        "Pharia-1-LLM-66b-control",
+    ]
+
+    def __init__(
+        self,
+        name: str = "Pharia-1-LLM-7b-control",
+        client: Optional[AlephAlphaClientProtocol] = None,
+    ) -> None:
+        super().__init__(name, client)
+
+    @property
+    def eot_token(self) -> str:
+        return "<|endoftext|>"
+
+    def to_chat_prompt(
+        self, messages: list[Message], response_prefix: str | None = None
+    ) -> RichPrompt:
+        return self.CHAT_PROMPT_TEMPLATE.to_rich_prompt(
+            messages=[m.model_dump() for m in messages], response_prefix=response_prefix
+        )
+
+
+class Llama3ChatModel(ChatModel):
+    """Instruction model to be used for `llama-3-*` and `llama-3.1-*` models.
+
+    Args:
+        name: The name of a valid llama-3 model.
+            Defaults to `llama-3-8b-instruct`
+        client: Aleph Alpha client instance for running model related API calls.
+            Defaults to :class:`LimitedConcurrencyClient`
+    """
+
+    CHAT_PROMPT_TEMPLATE = PromptTemplate(
+        """<|begin_of_text|>{% for message in messages %}<|start_header_id|>{{message.role}}<|end_header_id|>
+
+{% promptrange instruction %}{{message.content}}{% endpromptrange %}<|eot_id|>{% endfor %}<|start_header_id|>assistant<|end_header_id|>
```

Comment:
assistant determines the start of the generation?
I thoughts its 
System
User
Assistant
User 
Assistant
etc
But this is not how this template looks like?

Reply:
or is this "Chat" model for single turn conversation still?

Reply:
that's exactly what it is

And then we append assistant at the end to finally prompt the model